openstack/images_image_v2.go

```diff
@@ -372,3 +372,25 @@ func imagesFilterByProperties(v []images.Image, p map[string]string) []images.Im
 
 	return result
 }
+
+// dataSourceValidateImageSortFilter checks that the sorting filter passed
+// by a user follows the glance API definition of "sort_key:sort_dir,sort_key:sort_dir:.."
+// where sort_dir is optional.
+func dataSourceValidateImageSortFilter(v interface{}, k string) (ws []string, errors []error) {
+	validSortKeys := []string{"name", "owner", "protected", "status", "tag", "visibility",
+		"os_hidden", "member_status", "created_at", "updated_at"}
```

Comment:
It's better to enum these value on each line. This will reduce confusion on diff once new fields are added.

```suggestion
	validSortKeys := []string{
		"name",
		"owner",
		"protected",
		"status",
		"tag",
		"visibility",
		"os_hidden",
		"member_status",
		"created_at",
		"updated_at",
	}
```

In addition I think it makes sense to add a link to upstream OpenStack docs, e.g. https://docs.openstack.org/glance/latest/user/glanceapi.html#filtering-images-lists

Reply:
Done. 

> https://docs.openstack.org/glance/latest/user/glanceapi.html#filtering-images-lists

This link covers filtering and not sorting. I looked for the equivalent for sorting but couldn't find it. I have, instead, added the link to the glance api docs the cover sorting briefly.